src/components/SchemaViewer/index.jsx

```diff
@@ -1,34 +1,56 @@
-import React, { useState } from 'react';
-import { shape, string } from 'prop-types';
+import React, { useState, Fragment } from 'react';
+import Header from '../Header';
 import SchemaTable from '../SchemaTable';
+import SourceView from '../SourceView';
+import { schema } from '../../utils/prop-types';
 import { createSchemaTree } from '../../utils/schemaTree';
 
 function SchemaViewer({ schema }) {
   /**
    * Create a tree structure based on the given schema.
    * This acts as an intermediary data structure to define the overall
    * structure the schemaTable component will create based upon.
-   * It is managed as a state in order to handle the dynamic structure
-   * of $ref schemas so that when a cetain $ref is expanded or shrunk,
    * the entire tree structure will be updated to re-create and re-render
-   * the schema table.
+   * the schema table when a $ref is expanded or shrunk.
    */
   const [schemaTree, setSchemaTree] = useState(createSchemaTree(schema));
+  /**
+   * Track the mode of the schema viewer.
+   * By default, the mode is set to a table mode to display a schemaTable.
+   * If the mode is set to a source mode, displays the schema source instead.
+   */
+  const [sourceMode, setSourceMode] = useState(false);
+
+  function handleViewToggle() {
+    setSourceMode(prev => !prev);
+  }
 
-  return <SchemaTable schemaTree={schemaTree} setSchemaTree={setSchemaTree} />;
+  return (
+    <Fragment>
+      <Header
+        schema={schema}
+        sourceMode={sourceMode}
+        toggleMode={handleViewToggle}
+      />
+      {sourceMode ? (
+        <SourceView schema={schema} />
+      ) : (
+        <SchemaTable schemaTree={schemaTree} setSchemaTree={setSchemaTree} />
+      )}
+    </Fragment>
+  );
 }
 
 SchemaViewer.propTypes = {
   /** Schema input given to render */
-  schema: shape({
-    /** Type of schema */
-    type: string,
-  }),
+  schema: schema,
 };
 
 SchemaViewer.defaultProps = {
   /** Null type schema is set as default prop */
   schema: {
+    title: 'Empty title',
+    description: 'No description available',

```

Comment:
When the schema doesn't have a title/description, the storybook is currently not displaying anything.
![Screen Shot 2020-02-11 at 1 16 41 PM](https://user-images.githubusercontent.com/3766511/74265560-c299f680-4cd0-11ea-97b0-1f63847a382a.png)

Reply:
Fixed to displaying the following:

<img src="https://user-images.githubusercontent.com/29671309/74322963-19cec280-4dc8-11ea-9b7a-f4d19e429254.png" width="250px" />

Reply:
Thanks for fixing that. Let's default the title and description to empty strings for better UX.